--- a/PKGBUILD
+++ b/PKGBUILD
@@ -21,11 +21,18 @@

 makedepends=('meson' 'xcb-proto' 'mesa' 'libxslt' 'docbook-xsl' 'wayland-protocols')

-source=(https://waffle.freedesktop.org/files/release/waffle-${pkgver}/waffle-${pkgver}.tar.xz{,.asc})
+source=(https://waffle.freedesktop.org/files/release/waffle-${pkgver}/waffle-${pkgver}.tar.xz{,.asc}
+        "https://gitlab.freedesktop.org/mesa/waffle/-/merge_requests/158.diff")
 sha256sums=('f616c0ca6072d36bf81fb79a3695799973a978775e0213744bf0a39be6367425'
-            'SKIP')
+            'SKIP'
+            '585e1c67d7dc52619c8e66af8b94c7d0d69781132e2007bd1b7d89682bdab77a')
 validpgpkeys=('8703B6700E7EE06D7A39B8D6EDAE37B02CEB490D')

+prepare() {
+  cd "${srcdir}/waffle-${pkgver}"
+  patch -p1 < "${srcdir}"/158.diff
+}
+
 build() {
   local meson_args=(
     --buildtype release
